### ISSUE TITLE
Suggested change in error signalling

### ISSUE THREAD

**Issue by berendbaas** [miWebb/miBadger#3]
As it stands right now, most classes in miBadger return a true/false status to indicate wether an operation succeeded. While this is fine for trivial operations, in critical methods such as Settings::load, it would be better if the method couldn't fail silently, but would instead throw an exception, to make bugs more apparent to the programmer.

**Comment by rubentebogt** [comment]
@berendbaas Agreed, returning false will not trigger the error handler and will not be noticed upon. However, if you are always expecting to find a value then throw the exception if it is missing. The exception would mean that there was a problem.

If the value can be missing or present and both are valid for the application logic then return a null or false. More important: What do you do other places in the code? Consistency is important.
